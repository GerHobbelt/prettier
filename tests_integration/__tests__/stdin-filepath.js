"use strict";

const runPrettier = require("../runPrettier");

test("format correctly if stdin content compatible with stdin-filepath", () => {
  const result = runPrettier(
    "cli",
    ["--no-color", "--stdin-filepath", "abc.css"],
    { input: ".name { display: none; }" } // css
  );

  expect(result.stdout).toMatchSnapshot();
  expect(result.stderr).toMatchSnapshot();
  expect(result.status).toEqual(0);
});

test("throw error if stdin content incompatible with stdin-filepath", () => {
  const result = runPrettier(
    "cli",
    ["--no-color", "--stdin-filepath", "abc.js"],
    { input: ".name { display: none; }" } // css
  );

  expect(result.stdout).toMatchSnapshot();
  expect(result.stderr).toMatchSnapshot();
  expect(result.status).not.toEqual(0);
});
