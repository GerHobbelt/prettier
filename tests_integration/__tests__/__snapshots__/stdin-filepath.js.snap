// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format correctly if stdin content compatible with stdin-filepath (stderr) 1`] = `""`;

exports[`format correctly if stdin content compatible with stdin-filepath (stdout) 1`] = `
".name {
  display: none;
}
"
`;

exports[`format correctly if stdin content compatible with stdin-filepath (write) 1`] = `Array []`;

exports[`throw error if stdin content incompatible with stdin-filepath (stderr) 1`] = `
"stdin: SyntaxError: Unexpected token (1:1)
> 1 | .name { display: none; }
    | ^
"
`;

exports[`throw error if stdin content incompatible with stdin-filepath (stdout) 1`] = `""`;

exports[`throw error if stdin content incompatible with stdin-filepath (write) 1`] = `Array []`;
